packages/rxtranscribe/src/operators/bufferBetweenSilence.js

```diff
@@ -0,0 +1,112 @@
+import {of, iif, generate, EMPTY, forkJoin, BehaviorSubject} from 'rxjs';
+import {finalize, zip, takeLast, share, mergeMap, map, tap, pairwise, reduce, skipWhile, expand, takeWhile, takeUntil, take} from 'rxjs/operators';
+import toVAD, {VadEvents} from './toVAD';
+
+const MIN_CHUNK_SIZE = 10000;
+const VAD_SAMPLE_SIZE = 1000;
+
+/**
+ *
+ * @param vadOptions options that vad uses for the operator
+ * @param _toVAD The VAD operator
+ */
+const bufferBetweenSilence = function bufferBetweenSilence({
+  vadOptions = { sampleRate: 16000 },
+  _toVAD = toVAD,
+}) {
+  return chunk$ => {
+    let offset = 0;
+
+    // Replays the same chunk that was supplied
+    const chunkSub$ = chunk$.pipe(share());
+
+    const pump = (subject$) => ([vadRes, chunk]) => {
+      console.log('Offset', offset);
+      console.log('Chunk Length', chunk.length);
+      // console.log('Pump values', [vadRes, chunk]);
+      offset += VAD_SAMPLE_SIZE;
+      console.log('this should be false', chunk.length - offset < MIN_CHUNK_SIZE);
+      console.log('this should also be false', vadRes === VadEvents.SILENCE);
+      if (chunk.length - offset < MIN_CHUNK_SIZE || vadRes === VadEvents.SILENCE) {
+        console.log('We should not be getting here.');
+        return subject$.complete();
+      }
+      console.log('We should definitely get here.');
+      return subject$.next(offset);
+    };
+
+    const byteOffset$ = new BehaviorSubject(0);
+    const pumpNext = pump(byteOffset$);
+
+    // Creates slices for VAD to run against
+    const slicer$ = byteOffset$.pipe(
+      // tap(r => console.log('Inside the byteOffset', offset)),
+      zip(chunkSub$),
```

Comment:
interesting.  I did not know that zip could be used as an operator.  I've only ever used it as a generator like `const newObs$ = zip(obs1, obs2);`

Reply:
💡 okay...  I looked it up and the `zip` operator has been deprecated and replaced with a `zipAll` operator and `zip` index function (which creates a source observable).  That's apparently why I didn't recognize it...

Reply:
I'm not sure I follow what you mean by each "byteOffset corresponds 1:1"